rfcs/20221006-frontend_event_tracking.md

```diff
@@ -0,0 +1,469 @@
+    /**
+     * The name of the Vue component used to switch content types.
+     * @todo: We need to determine the best way of creating a string
+     * representation of a vue component, which allows us to know
+     * which ui element was clicked. Perhaps we could use unique data-test-ids
+     * or another unique identifier for this?
+     */
+    component: string;
+  };
+  /**
+   * Description: Whenever the user selects a result from the search results page.
+   * Questions:
+   *   - Which results are most popular for given searches?
+   *   - How often do searches lead to clicking a result?
+   *   - Are there popular searches that do not result in result selection?
+   */
+  SELECT_SEARCH_RESULT: {
+    /** If the result is a related result, provide the ID of the 'original' result */
+    related: string | null;
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The slug (not the prettified name) of the provider */
+    provider: string;
+    /** The search term */
+    query: string;
+  };
+  /**
+   * Description: The user clicks the CTA button to the external source to use the image
+   * Questions:
+   *   - How often do users go to the source after viewing a result?
+   */
+  GET_MEDIA: {
+    /** The unique ID of the media */
+    id: string;
+    /** The slug (not the prettified name) of the provider */
+    provider: string;
+    /** The media type being searched */
+    mediaType: MediaType;
+  };
+  /**
+   * Description: The user clicks one of the buttons to copy the media attribution
+   * Questions:
+   *   - How often do users use our attribution tool?
+   *   - Which format is the most popular?
+   */
+  COPY_ATTRIBUTION: {
+    /** The unique ID of the media */
+    id: string;
+    /** The format of the copied attribution */
+    format: "plain" | "rich" | "html";
+    /** The media type being searched */
+    mediaType: MediaType;
+  };
+  /**
+   * Description: The user reports a piece of media through our form
+   * Questions:
+   *   - How often do we get reports?
+   *   - Which types of reports are more common?
+   *   - Do we see an uptick in reports when a certain provider
+   *     is added/updated/refreshed?
+   */
+  REPORT_MEDIA: {
+    /** The unique ID of the media */
+    id: string;
+    /** The slug (not the prettified name) of the provider */
+    provider: string;
+    /** The media type being searched */
+    mediaType: MediaType;
+  };
+  /** Description: The user plays, pauses, or seeks an audio track.
+   * @todo: This potentially requires throttling.
+   *
+   * Questions:
+   *   - Do users interact with media frequently?
+   *   - Is it more common to playback audio on single results
+   *     or search pages?
+   *   - How many audio plays do we get?
+   */
+  AUDIO_INTERACTION: {
+    /** The unique ID of the media */
+    id: string;
+    event: "seek" | "play" | "pause";
+    /** The slug (not the prettified name) of the provider */
+    provider: string;
+  };
```

Comment:
I think it would be valuable to include the audio component in question.

```suggestion
  /**
   * Description: The user plays, pauses, or seeks an audio track.
   * @todo: This potentially requires throttling.
   *
   * Questions:
   *   - Do users interact with media frequently?
   *   - Is it more common to playback audio on single results
   *     or search pages?
   *   - How many audio plays do we get?
   */
  AUDIO_INTERACTION: {
    /** The unique ID of the media */
    id: string;
    event: "seek" | "play" | "pause";
    /** The slug (not the prettified name) of the provider */
    provider: string;
    /** The name of the Vue component used on the interaction, e.g. the global or main player. */
     component: string;
  };
```

Reply:
Fantastic! Added in 5ea8215.